lib/handlers/addCategory.ts

```diff
@@ -0,0 +1,19 @@
+import { CategoryService } from '../services/categoryService';
+import { CategoryRepository } from '../repositories/categoryRepository';
+import { Category } from "../models/category";
+
+interface AppSyncEvent {
+    arguments: {
+        [key: string]: any;
+    };
+}
+
+export async function main(event: AppSyncEvent): Promise<Category> {
+    const repo = new CategoryRepository();
+    const service = new CategoryService(repo);
+
+    const categoryData: Category =  event.arguments.category;
```

Comment:
We are not validating if we are receiving what we expect to receive!? i.e. `.parse()`, etc.

As this would be required for nearly all the handlers, a reusable zod service might make sense here, for validing/error handling.

![image](https://github.com/shahrukhkhan48/OnlineShop/assets/134911470/d03ea56c-d9c0-47ec-ab29-fe47a7ef747d)

Reply:
zod has been used in models